### PR TITLE
Moved filter parsing to library. Added check-ec2-filter

### PR DESCRIPTION
**Pull request by jmccarty3**
Move filter parsing logic into a separate library for reference.

Added a sensu check to provide warnings and critical alerts for a given filter.
Comparisons can be done via "equal, not, less, greater".

### CODE REVIEW

File: bin/check-ec2-filter.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: aws-sdk
+#   gem: sensu-plugin
+#
+# USAGE:
+#   ./check-ec2-filter.rb -f "{name:tag-value,values:[infrastructure]}"
+#   ./check-ec2-filter.rb -f "{name:tag-value,values:[infrastructure]} {name:instance-state-name,values:[running]}"
```
Comment (tas50) [inline bin/check-ec2-filter.rb]: Shouldn't the example contain the warn / critical flags?

File: bin/check-ec2-filter.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
+#! /usr/bin/env ruby
+#
+# check-ec2-filter
+#
+# DESCRIPTION:
+#   This plugin retrieves EC2 instances matching a given filter and
+#   returns the number matched
```
Comment (tas50) [inline bin/check-ec2-filter.rb]: This description should better explain that the check alerts on counts of instances returned.  This description sounds more like the metrics plugin

File: README.md [hunk 3]
```diff
@@ -17,6 +17,8 @@
 
 **check-ebs-snapshots.rb**
 
+**check-ec2-filter.rb**
```
Comment (tas50) [inline README.md]: This needs to get added to the files section as well.  Right now they're basically a dupe.  We need to go through at some point and add the descriptions of what the plugins do to the repo.

File: bin/check-ec2-filter.rb (hunk 2 above)
Comment (jmccarty3) [inline bin/check-ec2-filter.rb]: Updated to use a better description

File: bin/check-ec2-filter.rb (hunk 1 above)
Comment (jmccarty3) [inline bin/check-ec2-filter.rb]: Good call. Has been updated.

**Comment by jmccarty3** [comment]
Cleaned up the pull request.

**Comment by mattyjones** [comment]
@tas50  We good to merge this or do you want to f with it some more?

**Comment by jmccarty3** [comment]
@mattyjones @tas50 Were there any other issues with the merge?

**Pull request merged**